urbanopt_des/urbanopt_results.py

```diff
@@ -518,10 +550,23 @@ def process_results(self, building_names: list[str], year_of_data: int = 2017) -
         """
         # reset the data to None in case we are reprocessing
         self.data = None
+        # TODO: I think we should None out all of the data_* objects too
+
+        # reset the building characteristics
+        self.building_characteristics = {}
         for building_id in building_names:
-            print(f"Processing building {building_id}")
+            print(f"Reading building characteristics for {building_id}")
+            # read in the JSON file with the feature results, these are the building characteristics such
+            # as square footages, window areas, etc.
+            feature_json = self.get_urbanopt_default_feature_report_json(self.path / "run" / f"{self.scenario_name}" / f"{building_id}")
+            # Read the JSON as dictionary to the building characteristics
+            #  Use a context manager for opening files
+            with open(feature_json) as f:
+                self.building_characteristics[building_id] = json.load(f)
```

Comment:
FWIW read_text() does the open, read, close thing for files. It's one of my favorite one-liners. [Here](https://github.com/urbanopt/geojson-modelica-translator/blob/develop/geojson_modelica_translator/system_parameters/system_parameters.py#L964) is an example of mine in the GMT.

Reply:
love it